src/client/components/UserPage/Drawer/ControlPanel/helpers/reducers.ts

```diff
@@ -0,0 +1,48 @@
+export type State = {
+  controlPanelIsOpen: boolean
+  relevantShortLink?: string
+  qrCodeModalIsOpen: boolean
+}
+
+export const initialState: State = {
+  controlPanelIsOpen: false,
+  qrCodeModalIsOpen: false,
+}
+
+export type Action = {
+  type: string
+  payload?: string
+}
+
+export const DrawerActions = {
+  openControlPanel: 'OPEN_CONTROL_PANEL',
+  closeControlPanel: 'CLOSE_CONTROL_PANEL',
+  openQrCodeModal: 'OPEN_QR_CODE_MODAL',
+  closeQrCodeModal: 'CLOSE_QR_CODE_MODAL',
+}
+
+export function drawerReducer(state: State, action: Action) {
+  const newState: Partial<State> = {}
+
+  switch (action.type) {
+    case DrawerActions.openControlPanel:
+      newState.controlPanelIsOpen = true
+      newState.relevantShortLink = action.payload
+      break
+    case DrawerActions.closeControlPanel:
+      newState.controlPanelIsOpen = false
+      newState.relevantShortLink = undefined
```

Comment:
Nitpicking but should this be null instead? Since conceptually null means that there's no relevant link

Reply:
Converted optional string to nullable string in commit de093b3